lib/rules/newline-before-return.js

```diff
@@ -101,10 +102,18 @@ module.exports = function(context) {
      * @private
      */
     function hasNewlineBefore(node) {
-        var tokenBefore = sourceCode.getTokenBefore(node);
-        var lineNumTokenBefore = tokenBefore.loc.end.line;
-        var lineNumNode = node.loc.start.line;
-        var commentLines = calcCommentLines(node, tokenBefore);
+        var tokenBefore = sourceCode.getTokenBefore(node),
+            lineNumNode = node.loc.start.line,
+            lineNumTokenBefore,
+            commentLines;
+
+        if (tokenBefore) {
+            lineNumTokenBefore = tokenBefore.loc.end.line;
+        } else {
+            lineNumTokenBefore = 1;     // assume start of file
```

Comment:
Two things: 

1) I don't think this check needs to happen twice. You could also just pass `lineNumTokenBefore` to the `calcCommentLines` function.

2) More importantly, I think `lineNumTokenBefore = 1` is the wrong assumed value if we have encounter a global `return`. It leads to these cases passing (which they shouldn't, given the current intention of the rule):
```js

// comment
return;
```
and
```js

return;
```